### PR TITLE
feat(messages): filter messages by date

### PR DESCRIPTION
**Pull request by Doug-Reed**
MUMUP-3148: Filters messages for validity beteween goLive and expireDate

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: components/portal/messages/filters.js [hunk 1]
```diff
@@ -68,31 +68,41 @@ define(['angular', 'moment'], function(angular, moment) {
         });
       };
     })
-    .filter('filterByDate', function() {
```
Comment (vertein) [inline components/portal/messages/filters.js]: do we need to inject $filter?  looks like all the new code uses moment.

File: components/portal/messages/controllers.js [hunk 2]
```diff
@@ -95,20 +100,18 @@ define(['angular'], function(angular) {
...
+          $scope.seenMessageIds = result[0];
+          var dataMessages = result[1];
+          var groupedMessages = result[2];
+          var filteredMessages = $filter('filterForCommonElements')(
+            groupedMessages,
+            dataMessages
+          );
+          var dateFilteredMessages =
+            $filter('filterByDate')(filteredMessages);
+             $scope.messages =
```
Comment (vertein) [inline components/portal/messages/controllers.js]: Not a big deal, but the indentation of this line confused me quite a bit.  It took a while for me to figure out that line 112 was not part of the datFilteredMessages function.

File: components/portal/messages/filters.js [hunk 3]
```diff
@@ -68,31 +68,41 @@ define(['angular', 'moment'], function(angular, moment) {
...
+    .filter('filterByDate', ['$filter', function($filter) {
```
Comment (vertein) [inline components/portal/messages/filters.js]: do we need to inject $filter?  looks like all the new code uses moment.

**Comment by ChristianMurphy** [comment]
unrelated dependency CI issue logged as https://github.com/uPortal-Project/uportal-app-framework/issues/651

File: components/portal/messages/controllers.js [hunk 4]
```diff
@@ -51,42 +50,48 @@ define(['angular'], function(angular) {
...
-            // Call filtered notifications promises, then pass on to
-            // the completion function
-            $q.all(promiseFilteredMessages)
-              .then(filterMessagesSuccess)
-              .catch(filterMessagesFailure);
+
+          $q.all([promiseSeenMessageIds(),
+            promiseMessagesByData(allMessages),
+            groupPromise])
+            .then(filterMessagesSuccess)
```
Comment (ChristianMurphy) [inline components/portal/messages/controllers.js]: :art: It took me a solid minute to recognize that the `then` is not inside the `$q.all`. Maybe some indentation improvements could be made?

**Pull request merged**